scripts/entrypoint.sh

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/bash
+
+JSON_SETTINGS_TEMPLATE="/usr/bin/settings-tmpl.json"
+INDEX="/usr/share/nginx/html/index.html"
+
+# process and export json settings from given template (escape double quotes, remove spaces and line breaks)
+export LAUNCHER_JSON_SETTINGS="$(envsubst < ${JSON_SETTINGS_TEMPLATE} | tr -d '[:space:]' | sed 's/\"/\\\"/g')"
+
+echo LAUNCHER_JSON_SETTINGS=${LAUNCHER_JSON_SETTINGS}
+
+# create injected index.html with json settings
+echo $LAUNCHER_JSON_SETTINGS >> ${INDEX}
```

Comment:
having script outside of the `<html>` is really weird, are u sure it's working (and that it get executed before the react app)?

Reply:
We know because of this: https://github.com/fabric8-launcher/launcher-creator-welcome-app/blob/449fea74814b4c4d1cee463b6fb7cf4ffc19aaa5/Dockerfile#L12

Reply:
So it's actually the very first thing that gets executed, before nginx, before anything else

Reply:
@quintesse I am talking about the app in the browser :) have you looked at the generated `index.html`?
```html
<html>...</html>
<script>..</script>
```

Reply:
The `<script>` which set envs have to be executed before the React app does

Reply:
I didn't test because the app still waits for `process.env.REACT_APP_WELCOME_FILE` in the sources

Reply:
@gastaldi could you inject the script tag just before the end of the `</head>` tag

Reply:
In the mean time I am taking care of the app itself..